src/server/api/routers/team/utils.ts

```diff
@@ -0,0 +1,108 @@
+import {TRPCError} from "@trpc/server";
+import {type Context} from "~/server/api/trpc";
+import {type JoinUserTeamInput, type LeaveUserTeamInput} from "~/server/api/routers/team/types";
+
+
+export async function userSwitchTeamTransaction(
+  ctx: Context,
+  input: JoinUserTeamInput | LeaveUserTeamInput
+) {
+  const userInformation = await ctx.db.userInformation.findFirst({
+    where: {
+      userId: input.userId
+    },
+  })
+
+  if (!userInformation) {
+    throw new TRPCError({
+      code: 'NOT_FOUND',
+      message: `Could not find user with id ${input.userId}`,
+    });
+  }
+
+  const currentTeamId = userInformation.teamId;
+
+  await ctx.db.$transaction(async (prisma) => {
+
+    // Find the team to leave
+    const teamToLeave = await ctx.db.team.findFirst({
+      where: {
+        id: currentTeamId
+      }
+    })
+
+    if (!teamToLeave) {
```

Comment:
Would recommend adding `TODO` comments to add error boundaries for each of these, so that if one front-end component that uses this fails it won't cascade up the whole tree and crash the entire app. 

We probably want to show these messages in modals or toasts to the user. There should be a descriptive version of the message as well that will allow you to insta-debug under pressure.